.github/workflows/tests.yml

```diff
@@ -0,0 +1,185 @@
+name: tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  linux:
+
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        php: ["5.6", "7.0", "7.1", "7.2", "7.3", "7.4"]
+        stability: ["Stable"]
+        include:
+          - php: "5.6"
+            stability: "Lowest"
+          - php: "8.0"
+            stability: "Ignore"
+
+    name: PHP ${{ matrix.php }} - ${{ matrix.stability }} - Linux
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools: composer:snapshot
+          coverage: none
+
+      - name: Setup problem matchers
+        run: echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+      - name: Install platform ignoring dependencies
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer update --prefer-dist --no-interaction --no-progress --ignore-platform-req=php --ignore-platform-req=hhvm
+        if: "matrix.stability == 'Ignore'"
+
+      - name: Install stable dependencies
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer update --prefer-dist --no-interaction --no-progress --ignore-platform-req=hhvm
+        if: "matrix.stability == 'Stable'"
+
+      - name: Install lowest dependencies
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer update --prefer-stable --prefer-lowest --prefer-dist --no-interaction --no-progress --ignore-platform-req=hhvm
+        if: "matrix.stability == 'Lowest'"
+
+      - name: Execute tests
+        run: vendor/bin/phpunit
+        if: "matrix.php != '8.0'"
+
+      - name: Execute tests with weak deprecation warnings
+        run: SYMFONY_DEPRECATIONS_HELPER=weak vendor/bin/phpunit
+        if: "matrix.php == '8.0'"
+
+      - name: Execute php-cs-fixer standard mode
+        run: PHP_CS_FIXER_FUTURE_MODE=1 php php-cs-fixer --diff --dry-run -v fix
+        if: "matrix.php != '8.0'"
+
+      - name: Execute php-cs-fixer ignoring env
+        run: PHP_CS_FIXER_FUTURE_MODE=1 PHP_CS_FIXER_IGNORE_ENV=1 php php-cs-fixer --diff --dry-run -v fix
+        if: "matrix.php == '8.0'"
+
+  windows:
```

Comment:
Why do we need this? Why OS cannot be part of matrix?

Reply:
Each OS is different, and in the end, trying to put it all into one single thing ends up an absolute mess. Sometimes duplication is least worst. Certainly makes the config file more readable.

Reply:
not sure, here it was added without that "absolute mess"
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5268/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR33